example/follow_me/fake_location_provider.cpp

```diff
@@ -0,0 +1,60 @@
+
+#include "fake_location_provider.h"
+
+void FakeLocationProvider::request_location_updates(location_callback_t callback)
+{
+    location_callback_ = callback;
+    timer_.async_wait(boost::bind(&FakeLocationProvider::compute_next_location, this));
```

Comment:
Do you really really need boost for this? I tried to keep everything to plain C++11, if at all possible, I'd vote to do this without boost.

I mean do you actually need to do it async? Or could you just use a loop where you set the location?

Reply:
In my opinion the example doesn't need to show how you could/should build this with async. It should only show how to use the DroneCore API. The simpler the better.

Reply:
@julianoes A simple example in this case can show how the API is called, but now how it *should* be used.  I think the way it is now will save readers time and energy. Your call though.

Reply:
Thanks for the review @julianoes @hamishwillee. As discussed earlier our example should help application developers including Android, Windows, etc; I simulated approximate layout for the name. I tried simulating [Android Location Provider](https://developer.android.com/training/location/receive-location-updates.html#updates): `requestLocationUpdates`.
Even [Apple](https://developer.apple.com/documentation/corelocation/cllocationmanager/1423750-startupdatinglocation)  and [Windows](https://docs.microsoft.com/en-us/uwp/api/Windows.Devices.Geolocation.Geolocator#Examples) use this style. As mentioned earlier, I thought using Boost is portable way. I will try to check whether it can be done without Boost.